e2e/testclients/clientutils/util.go

```diff
@@ -44,55 +44,70 @@ type NotificationItem struct {
 	Persistent bool      `json:"persistent"`
 }
 
-type Content struct {
+type Event struct {
 	Message string `json:"message"`
 }
 
+type Receipt struct {
+	TxHash string         `json:"txHash"`
+	Result map[string]any `json:"result"`
+	Errors []string       `json:"errors"`
+}
+
 type NotificationCollection struct {
 	Notifications   []NotificationItem `json:"notifications"`
 	CacheableCursor string             `json:"cacheableCursor"`
 }
 
-func (c *NakamaClient) ListNotifications(k int) ([]*Content, error) {
+func (c *NakamaClient) ListNotifications(k int) ([]*Receipt, []*Event, error) {
 	path := "v2/notification"
```

Comment:
Function signature is a bit busy here. It returns three values. Better to split this into two functions. ListEvents and ListReceipts, with ListNotifications as an underlying base call and the filtering for subject line happening in ListEvents and ListReceipts.